lib/core/connect.js

```diff
@@ -12,30 +12,30 @@ const { InvalidArgumentError } = require('./errors')
 // re-use is enabled.
 
 class Connector {
-  constructor ({ tls, socketPath, maxCachedSessions }) {
-    this.tls = tls || {} // TODO: Make shallow copy to protect against mutations.
-
+  constructor ({ maxCachedSessions, socketPath, ...opts }) {
     if (maxCachedSessions != null && (!Number.isInteger(maxCachedSessions) || maxCachedSessions < 0)) {
       throw new InvalidArgumentError('maxCachedSessions must be a positive integer or zero')
     }
 
-    this.socketPath = socketPath
+    this.opts = { path: socketPath, ...opts }
```

Comment:
Wouldn't it be faster to:

```js
opts.path = socketPath;
this.opts = opts;
```

Reply:
/cc @ronag

Reply:
We should avoid mutating user owned objects. I don’t think this is a hot path.